tools/test_arm.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+
+Copyright (c) 2012-2016, Arvid Norberg, Alden Torres
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in
+      the documentation and/or other materials provided with the distribution.
+    * Neither the name of the author nor the names of its
+      contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+#include "libtorrent/config.hpp"
+#include "libtorrent/sha1_hash.hpp"
+#include "libtorrent/hex.hpp" // from_hex
+
+#include <vector>
+#include <iostream>
+
+using namespace libtorrent;
+
+static sha1_hash to_hash(char const* s)
+{
+	sha1_hash ret;
+	aux::from_hex(s, 40, (char*)&ret[0]);
+	return ret;
+}
+
+int main(int argc, char* argv[])
+{
+	std::cout << "running test arm" << std::endl;
+
+	bool has_builtin_clz =
+#if TORRENT_HAS_BUILTIN_CLZ
+		true;
+#else
+		false;
+#endif
+	TORRENT_ASSERT(has_builtin_clz);
```

Comment:
if you want to assert a compile time property, you could just do:

#ifndef TORRENT_HAS_BUILTIN_CLZ
#error "expected built-in clz"
#endif